hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/JdbcSqlConnector.java

```diff
@@ -375,56 +385,117 @@ public Vertex updateProcessor(
-        assert predicate == null;
+    public Vertex deleteProcessor(
+            @Nonnull DagBuildContext context,
+            @Nullable HazelcastRexNode predicate,
+            boolean hasInput
+    ) {
+        // We don't allow both predicate and input at the same time
+        assert predicate == null || !hasInput;
+
         JdbcTable table = context.getTable();
 
-        List<String> pkFields = getPrimaryKey(context.getTable())
-                .stream()
-                .map(f -> table.getField(f).externalName())
-                .collect(toList());
+        RexNode rexPredicate = predicate == null ? null : predicate.unwrap(RexNode.class);
 
-        DeleteQueryBuilder builder = new DeleteQueryBuilder(table, pkFields, resolveDialect(table, context));
-        return context.getDag().newUniqueVertex(
-                "Delete(" + table.getExternalNameList() + ")",
-                new DeleteProcessorSupplier(
-                        table.getDataConnectionName(),
-                        builder.query(),
-                        table.getBatchLimit()
+        DeleteQueryBuilder builder = new DeleteQueryBuilder(table,
+                resolveDialect(table, context),
+                rexPredicate,
+                hasInput
+        );
+
+        DmlProcessorSupplier deletePS = new DmlProcessorSupplier(
+                table.getDataConnectionName(),
+                builder.query(),
+                builder.dynamicParams(),
+                builder.inputRefs(),
+                table.getBatchLimit()
+        );
+
+        return dmlVertex(context, hasInput, table, deletePS, "Delete");
+    }
+
+    private static Vertex dmlVertex(
+            DagBuildContext context,
+            boolean hasInput,
+            JdbcTable table,
+            DmlProcessorSupplier processorSupplier,
+            String statement
+    ) {
+
+        if (!hasInput) {
+            // There is no input, we push the whole update query to the database, but we need single dummy item
+            // to execute the update in WriteJdbcP
+            // We can consider refactoring the WriteJdbcP, so it doesn't need the dummy item in the future.
+
+            // Use local member address to run the dummy source and processor on the same member
+            Address localAddress = context.getNodeEngine().getThisAddress();
+
+            Vertex v = dummySourceVertex(context, "DummySourceFor" + statement, localAddress);
+            Vertex updateVertex = context.getDag().newUniqueVertex(
+                    statement + "(" + table.getExternalNameList() + ")",
+                    forceTotalParallelismOne(processorSupplier, localAddress)
+            );
+
+            context.getDag().edge(Edge.between(v, updateVertex));
+            return updateVertex;
+        } else {
+            Vertex updateVertex = context.getDag().newUniqueVertex(
+                    statement + "(" + table.getExternalNameList() + ")",
+                    processorSupplier
```

Comment:
the way in which we create PMS does not obey https://docs.hazelcast.com/hazelcast/5.2/security/native-client-security#connector-permission. See `WriteJdbcP.metaSupplier`. We did not also before but maybe we should?